pkg/util/vsphere/watcher/watcher.go

```diff
@@ -291,27 +356,65 @@ func Start(
 	// Update the context with this watcher.
 	setContext(ctx, w)
 
-	var cancel context.CancelFunc
+	var (
+		cancel  context.CancelFunc
+		version string
+	)
+
 	ctx, cancel = context.WithCancel(ctx)
 	w.cancel = cancel
 
 	go func() {
 		defer func() {
+			logger.Info("Exiting watching VMs", "version", version)
+
 			// Remove this watcher from the context. While there is no watcher
 			// in the context, calls to Add/Remove will fail.
 			setContext(ctx, nil)
 
 			w.close()
 		}()
 
-		if err := w.pc.WaitForUpdatesEx(
-			ctx,
-			&property.WaitOptions{},
-			func(ou []vimtypes.ObjectUpdate) bool {
-				return w.onUpdate(ctx, ou)
-			}); err != nil {
+		req := vimtypes.WaitForUpdatesEx{
+			This: w.pc.Reference(),
+			Options: &vimtypes.WaitOptions{
+				MaxWaitSeconds:   ptr.To[int32](60 * 5),
+				MaxObjectUpdates: 100,
+			},
+			Version: version,
+		}
 
-			w.setErr(err)
+		for {
+			logger.V(4).Info("Waiting for updates while watching VMs",
+				"version", version)
```

Comment:
Not a big deal but how about using `req.Version` here and get rid the intermediate `version`? Only mentioned something b/c I had to do a double take a to what `req.Version` was being initialized to